snappy/handlers.go

```diff
@@ -48,10 +47,32 @@ func (h *Handler) setClient(c SnapdClient) {
 	h.snapdClient = c
 }
 
-func (h *Handler) getAll(w http.ResponseWriter, r *http.Request) {
+func (h *Handler) jsonResponseOrError(v interface{}, w http.ResponseWriter) {
 	w.Header().Set("Content-Type", "application/json")
 	enc := json.NewEncoder(w)
 
+	if err := enc.Encode(v); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "Error: %s", err)
+		log.Print(err)
+	}
+}
+
+func (h *Handler) snapOperationResponse(name string, err error, w http.ResponseWriter) {
+	msg := "Accepted"
+	status := http.StatusAccepted
+
+	if err != nil {
+		msg = "Processing error"
+		status = http.StatusInternalServerError
```

Comment:
Previously the calls to add and remove snaps were synchronous so we knew in-request the status of the operation. Now a background task is created so we can only say wether the operation was accepted or there was an error.